pkg/config/leaderelection/leaderelection.go

```diff
@@ -29,8 +29,20 @@ import (
 // See https://github.com/kubernetes/kubernetes/issues/107454 for
 // details on how to migrate to "leases" leader election.
 // Don't forget the callbacks!
-// TODO: In the next version we should switch to using "leases"
```

Comment:
@dgrisonnet and @tkashem I moved the TODO to where the logic for the default resource lock is made. Is this okay with you?

Reply:
Yes, actually, that was the goal

we want the following:
- A: a brand new operator being written should us `lease` (preferably by default)
- B: existing operators who are currently using `configmapsleases` should use `lease` (preferably by default)
- C: existing operators who have not yet upgraded libray-go for a while (still using `configmaps`) should start using `configmapsleases` as soon as they upgrade to the latest library-go (we can't have these operators use `lease` by default, it will break their upgrade flow)

I am concerned about `C` - if we don't have any existing operators in `C` category today then we can safely rename `ToLeaderElectionWithConfigmapLease` to `ToLeaderElectionWithLease` and have `lease` be the default option. 
So I would like to know if `C` is empty today - it can be possible today to count C, maybe by having a test in CI?

or is there a way we can break (static, compile time) these operators in C when they upgrade library-go so they can opt-in? I would like us to be able to avoid these changes if possible, especially if C is empty.

Reply:
Leases are not 100% reliable today.  It's been a while since the switch I think, I'd simply switch.

Reply:
@mprahl can you revise the pr to switch to `lease`? (disregard `C`, that means no need to make anything configurable)

Reply:
@tkashem in ACM we would like to still have the option for configmapsleases if that's okay. I'd make `lease` the default though.

Reply:
@mprahl  yeah, sure, go ahead. (just curious, you have an operator that is currently using configmaps?)